packages/block-library/src/navigation/edit/index.js

```diff
@@ -356,6 +355,7 @@ function Navigation( {
 											onClose();
 										} }
 										onCreateNew={ () => {
+											replaceInnerBlocks( clientId, [] );
```

Comment:
I added this because when you clicked "Create new menu" it would use the current inner blocks (i.e. from the current menu) and automatically create a new menu from those.

What we want is actually to remove all the inner blocks and reset in order that the placeholder will be shown.